Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/Pantograph.cs

```diff
@@ -342,20 +347,30 @@ public void HandleEvent(PowerSupplyEvent evt)
                             default:
                             case 1:
                                 soundEvent = Event.Pantograph1Up;
+                                Confirm(CabControl.Pantograph4, CabSetting.On);
                                 break;
 
                             case 2:
                                 soundEvent = Event.Pantograph2Up;
+                                Confirm(CabControl.Pantograph2, CabSetting.On);
                                 break;
 
                             case 3:
                                 soundEvent = Event.Pantograph3Up;
+                                Confirm(CabControl.Pantograph3, CabSetting.On);
                                 break;
 
                             case 4:
                                 soundEvent = Event.Pantograph4Up;
+                                Confirm(CabControl.Pantograph4, CabSetting.On);
                                 break;
                         }
+
+
+                        if (!Simulator.TRK.Tr_RouteFile.Electrified)
+                            Simulator.Confirmer.Warning(Simulator.Catalog.GetString("No power line!"));
```

Comment:
I'm probably being ambitious here, but: If we ever get our own route editor, this logic may need to be changed. We can expect OR to support electrified and non-electrified sections on the same route.

Reply:
This code was moved from a MSTSElectricLocomotive function:
public override void SignalEvent(PowerSupplyEvent evt, int id) (function has been removed)